src/App.js

```diff
@@ -8,51 +8,66 @@ class App extends React.Component {
     super(props)
 
     this.state={
-      list:List
+      list:List,
+      newTaskName:''
     }
-
-    this.add=this.add.bind(this)
-    this.clear=this.clear.bind(this)
-    this.edit=this.edit.bind(this)
   }
   // you will need a place to store your state in this component.
   // design `App` to be the parent component of your application.
   // this component is going to take care of state, and any change handlers you need to work with your state
   render() {
     return (
       <div>
-        <h2>Welcome to your Todo App!</h2>
         <TodoList
           list={this.state.list}
           add={this.add}
           clear={this.clear}
           edit={this.edit}
+          formChange={this.formChange}
+          newTaskName={this.state.newTaskName}
         />
       </div>
     );
   }
-
-  add(e){
+  
+  componentDidMount(){
+    if(localStorage.getItem('tasks'))
+      this.setState({list:JSON.parse(localStorage.getItem('tasks'))})
+  }
+  componentDidUpdate(){
+    localStorage.setItem('tasks',JSON.stringify(this.state.list))
+  }
```

Comment:
Interesting to see yo handling the local state with this logic here, This is definitely one way to skin that cat.